src/components/admin/article/CreateArticleController.jsx

```diff
@@ -0,0 +1,241 @@
+  buildPropMerger,
+} from 'components/hocs/falcor-hocs';
+import { withModals } from 'components/admin/hocs/modals/withModals';
+
+// Misc.
+import { logger } from 'lib/logger';
+
+// Define data needed by component
+const falcorPaths = [['categories', 'byIndex', { length: 30 }, ['name', 'id']]];
+
+const propMerger = buildPropMerger((data, currentProps) => {
+  const categories = parseFalcorPseudoArray(data.categories.byIndex).filter(
+    category => _.get(category, 'id') && _.get(category, 'name'),
+  );
+  return {
+    ...currentProps,
+    categories,
+  };
+});
+
+class CreateArticleController extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      slug: '',
+      authors: [],
+      tags: [],
+      teaser: '',
+      category: null,
+      imageUrl: '',
+      saving: false,
+    };
+  }
+
+  updateTitle = title => this.setState({ title });
+  updateSlug = slug => this.setState({ slug });
+  updateAuthors = authors => this.setState({ authors });
+  updateTags = tags => this.setState({ tags });
+  updateTeaser = teaser => this.setState({ teaser });
+  updateImage = imageUrl => this.setState({ imageUrl });
+  updateCategory = category => this.setState({ category });
+
+  handleDialogClose = () => {
+    browserHistory.push(getArticleListPath(this.props.params.page));
+  };
+
+  validateArticle = async () => {
+    // Check that it has all the required fields
+    const requiredFields = ['title', 'slug'];
+    const errorMessage = requiredFields
+      .map(key => {
+        if (!this.state[key]) {
+          return `${key} is a required field`;
+        }
+        return null;
+      })
+      .filter(x => x !== null)
+      .join('\n');
+    if (errorMessage) {
+      await this.props.displayAlert(errorMessage);
+      return false;
+    }
+    return true;
+  };
+
+  handleCreateArticle = async () => {
+    if (!await this.validateArticle()) {
+      return;
+    }
+
+    const createArticleArgument = _.pick(this.state, [
+      'title',
+      'slug',
+      'authors',
+      'tags',
+      'teaser',
+      'category',
+      'imageUrl',
+    ]);
+
+    let newArticle;
+    this.setState({ saving: true });
+    try {
+      const jsonGraph = await this.props.falcor.call(
+        ['articles', 'createNew'],
+        [createArticleArgument],
+      );
+      const createdArticles = parseFalcorPseudoArray(
+        jsonGraph.json.articles.byId,
+      );
+      if (createdArticles.length !== 1) {
+        logger.error(new Error('Expected exactly one article to be created'));
+        await this.props.displayAlert(buildErrorMessage());
+        return;
+      }
+      [newArticle] = createdArticles;
+    } catch (e) {
+      logger.error(e);
+      await this.props.displayAlert(buildErrorMessage());
+      return;
+    } finally {
+      // Finally runs even if we return from the function in try and catch
+      setTimeout(() => this.setState({ saving: false }), 100 * 1000);
```

Comment:
Nice one :P, that was indeed from testing, there shouldn't be a setTimeout at all haha, didn't catch it because when it doesn't error the url changes which unmounts the whole component